File: longestConsecutive.js
var longestConsecutive = function (nums) {
  if (nums.length < 1) {
    return 0;
  }
  nums = nums.sort(function (a, b) {
    return a - b;
  });
  let sum = 1;
  let results = [];

  for (let i = 1; i < nums.length; i++) {
    if (nums[i] - 1 === nums[i - 1]) {
      sum++;
    } else if (nums[i] === nums[i - 1]) {
      sum = sum;
    } else {
      results.push(sum);
      sum = 1;
    }
  }
  results.push(sum);
  return Math.max(...results);
};

File: python/longest_consecutive.py
def longest_consecutive(nums: list[int]) -> int:
    if len(nums) < 1:
            return 0
    nums.sort()
    sum = 1
    results = []
    for i in range(1, len(nums)):
        if nums[i] - 1 == nums[i-1] :
            sum += 1
        elif nums[i] == nums[i-1]:
            sum = sum
        else:
            results.append(sum)
            sum = 1
    results.append(sum)
    return max(results) 

